### PR TITLE
Use in-focus window as reference so multi-client users can have map o…

### PR DESCRIPTION
**Pull request by Farmith**
Small update that allows users with a multi-client setup to have the map on all their clients for ease

### CODE REVIEW

File: Helpers/GameMemory.cs [hunk 1]
```diff
@@ -48,7 +48,15 @@ public static GameData GetGameData()
             {
                 Process[] process = Process.GetProcessesByName(ProcessName);
                 Process gameProcess = process.Length > 0 ? process[0] : null;
-
+                
+                // Check which GameClient is in focus, and use that for the Overlay
+                foreach (Process p in process)
+                {
+                    if (WindowsExternal.GetForegroundWindow() == p.MainWindowHandle)
```
Comment (AudriusButkevicius) [inline Helpers/GameMemory.cs:55]: Call WindowsExternal.GetForegroundWindow() once before the loop maybe?

**Comment by OneXDeveloper** [comment]
A small issue with this is that it doesn't appropriately bust caches/detect game has changed when changing to the other players. For example, if all the characters join the same game and just walk outside of town (not via wp) and move around, you'll notice it'll be pinned to the first character and not show as moving when you move the other characters.

File: Helpers/GameMemory.cs (hunk 1 above)
Comment (Farmith) [reply Helpers/GameMemory.cs]: Added this, good catch!

**Comment by Farmith** [comment]
Added cache-invalidating procedures as per OneX request

**Review by AudriusButkevicius — APPROVED** [review]
Lgtm

**Comment by derekShaheen** [comment]
Looks like we've made very similar changes, but gone about it in slightly different ways.

**Comment by misterokaygo** [comment]
#123 will be addressing these changes.

**Comment by misterokaygo** [comment]
Thank you for your contribution.

**Pull request closed without merging**